src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -156109,5 +156109,161 @@ void F(object? o)
                 Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "stackalloc[] { x => x }").WithLocation(9, 18)
                 );
         }
+
+        [Fact, WorkItem(61361, "https://github.com/dotnet/roslyn/issues/61361")]
+        public void NullableBoolInVariousTests()
+        {
+            var comp = CreateCompilation(@"
+#nullable enable
+class C
+{
+    public static void M(bool? x)
+    {
+        if (x) // 1
+        {
+            x.Value.ToString(); // 2
+        }
+
+        _ = x ? x.Value.ToString() : string.Empty; // 3, 4
+
+        if (x == true)
+        {
+            x.Value.ToString();
+        }
+
+        _ = (x == true) ? x.Value.ToString() : string.Empty;
+
+        _ = x switch
+        {
+            true => x.Value.ToString(),
+            _ => string.Empty
+        };
+    }
+}
+");
+
+            comp.VerifyDiagnostics(
+                // (7,13): error CS0266: Cannot implicitly convert type 'bool?' to 'bool'. An explicit conversion exists (are you missing a cast?)
+                //         if (x) // 1
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "x").WithArguments("bool?", "bool").WithLocation(7, 13),
+                // (9,13): warning CS8629: Nullable value type may be null.
+                //             x.Value.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "x").WithLocation(9, 13),
+                // (12,13): error CS0266: Cannot implicitly convert type 'bool?' to 'bool'. An explicit conversion exists (are you missing a cast?)
+                //         _ = x ? x.Value.ToString() : string.Empty; // 3, 4
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "x").WithArguments("bool?", "bool").WithLocation(12, 13),
+                // (12,17): warning CS8629: Nullable value type may be null.
+                //         _ = x ? x.Value.ToString() : string.Empty; // 3, 4
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "x").WithLocation(12, 17)
+                );
+        }
+
+        [Theory, WorkItem(61361, "https://github.com/dotnet/roslyn/issues/61361")]
+        [InlineData("<")]
+        [InlineData("<=")]
+        [InlineData(">")]
+        [InlineData(">=")]
+        public void LiftedRelationalOperation(string op)
```

Comment:
consider accessing .Value in both branches of the `if`/`?:` so that we see the diagnostics for the "may be null" case.